spyder/utils/programs.py

```diff
@@ -71,17 +71,50 @@ def get_temp_dir(suffix=None):
 def is_program_installed(basename):
     """
     Return program absolute path if installed in PATH.
-
-    Otherwise, return None
-
-    On macOS systems, a .app is considered installed if
-    it exists.
+    Otherwise, return None.
+
+    Also searches specific platform dependent paths that are not already in
+    PATH. This permits general use without assuming user profiles are
+    sourced (e.g. .bash_Profile), such as when login shells are not used to
+    launch Spyder.
+
+    On macOS systems, a .app is considered installed if it exists.
+
+    Darwin:
+        /usr/local/bin
+        ~/opt/anaconda3/condabin
+        ~/opt/miniconda3/condabin
+        /opt/anaconda3/condabin
+        /opt/miniconda3/condabin
+    Linux:
+        ?
+    Windows:
+        ?
     """
-    if (sys.platform == 'darwin' and basename.endswith('.app') and
-            osp.exists(basename)):
-        return basename
+    home = get_home_dir()
+    req_paths = []
+    if sys.platform == 'darwin':
+        if basename.endswith('.app') and osp.exists(basename):
+            return basename
+
+        # prioritize anaconda before miniconda; local before global
+        req_paths.extend([
+            osp.join('/usr', 'local', 'bin'),
+            osp.join(home, 'opt', 'anaconda3', 'condabin'),
+            osp.join(home, 'opt', 'miniconda3', 'condabin'),
+            osp.join('/opt', 'anaconda3', 'condabin'),
+            osp.join('/opt', 'miniconda3', 'condabin')
+        ])
+    elif sys.platform.startswith('linux'):
+        # TODO: what are the possible conda paths?
+        # TODO: what are the possible pyenv paths?
+        pass
+    elif WINDOWS:
+        # TODO: what are the possible conda paths?
+        # TODO: what are the possible pyenv paths?
+        pass

```

Comment:
@dalthviz, could you provide the default installation locations for Anaconda and Miniconda on Windows here?

Reply:
For Anaconda:

* Installations for just the current user: the path is something like `C:\Users\<user>\anaconda3\condabin`
* Installations for all the user: the default path is something like `C:\ProgramData\Anaconda3\condabin`

For Miniconda:

* Installations for just the current user: the path is something like `C:\Users\<user>\miniconda3\condabin`
* Installations for all the user: the default path is something like `C:\ProgramData\Miniconda3\condabin`

Reply:
Looks like standard installation location for `pyenv` is `$HOME\.pyenv\pyenv-win\bin` and for `conda` is 
`$HOME\Anaconda3\condabin`
`$HOME\Miniconda3\condabin`
`C:\Anaconda3\condabin`
`C:\Miniconda3\condabin`
`C:\ProgramData\Anaconda3\condabin`
`C:\ProgramData\Miniconda3\condabin`

@dalthviz, can you confirm? Additions? Subtractions?

Reply:
LGTM :+1: